### ISSUE TITLE
[Adjust Rule] SIM401 False positive, rule too greedy

### ISSUE THREAD

**Issue by joaoe** [MartinThoma/flake8-simplify#177]
## Desired change

* **Rule(s)**: SIM401 
* **Adjustment**: The rule is too greedy and will happily trigger in code that has nothing to do with the supposed suggested simplification in the documentation
* 
## Example

This testcase shows the false posivie

```python
all_responses: dict[str, object] = {}
keep_responses = set()

key = "value"
if key in keep_responses:
    retval = all_responses[key]
else:
    retval  = all_responses.pop(key)
```
Output
```
sim401_bug.py:5:1: SIM401 Use 'retval = keep_responses.get(key, all_responses.pop(key))' instead of an if-block
```

Issues with the rule
1. keep_responses is not a dict, it is just a generic container.
2. the variable accessed inside the if block is not the same one in the condition
3. the else block is non-trival code with side-effects

## Suggestion

The rule should only trigger in the very strict case
```python
if key in container:
    retval = container[key]
else:
    retval  = "value"
```

Where
1. `container` must be accessed in both the condition with the form `key in container` and then the if block `container[key]`
2. `key` must be reused both in the condition and if block
3. `"value"` must be a literal or simple variable name e.g. `"value"` or `value`. Else, stuff like `value.property` or `value()` can have side-effects which change the state of the application.
